README.Rmd

```diff
@@ -136,43 +130,30 @@ percentile_wages %>%
     scale_y_continuous(labels = scales::dollar) +
     coord_flip() +
     theme_cmap() +
-    cmap_color_continuous(palette = "seq_red_purple")
+    cmap_color_continuous(palette = "multi_red_purple")
 ```
 
 #### Single-hue sequential gradients
 
 ```{r echo=FALSE, out.height="40%", out.width="50%"}
-viz_gradient(cmap_gradients$seq_reds)
-viz_gradient(cmap_gradients$seq_oranges)
-viz_gradient(cmap_gradients$seq_yellows)
-viz_gradient(cmap_gradients$seq_greens)
-viz_gradient(cmap_gradients$seq_teals)
-viz_gradient(cmap_gradients$seq_blues)
-viz_gradient(cmap_gradients$seq_purples)
-viz_gradient(cmap_gradients$seq_grays)
+cmap_gradients %>% 
+  keep(str_detect(names(.), "^seq")) %>% 
+  walk2(., names(.), viz_gradient)
 ```
 
 #### Multi-hue sequential gradients
 
 ```{r echo=FALSE, out.height="40%", out.width="50%"}
-viz_gradient(cmap_gradients$seq_yellow_orange_red)
-viz_gradient(cmap_gradients$seq_green_teal_blue)
-viz_gradient(cmap_gradients$seq_yellow_orange)
-viz_gradient(cmap_gradients$seq_orange_red)
-viz_gradient(cmap_gradients$seq_red_purple)
-viz_gradient(cmap_gradients$seq_yellow_green)
-viz_gradient(cmap_gradients$seq_green_teal)
-viz_gradient(cmap_gradients$seq_teal_blue)
+cmap_gradients %>% 
+  keep(str_detect(names(.), "^multi")) %>% 
```

Comment:
Can we revert the names and distinguish between single/multi hue gradients with a more nuanced regex? Specifically, multi-hues have more than 1 underscore, whereas single-hues only have one. I think the regex `^seq_[^_]+$` would work for single-hues and `^seq_[^_]+_.+$` would work for multi-hues.

Reply:
Sure...I did consider a more complicated regex, but counting underscores seemed like a less logical and easier to accidentally break solution. This isn't really important to me either way, but why not have a distinguishing prefix?

Reply:
When I named them I felt it was important to distinguish between sequential and diverging (i.e. the gradients' use cases) rather than number of hues, because the diverging gradients are all multi-hue as well, so this could add confusion.

If you're worried about something breaking (which shouldn't happen as this list of gradients will be pretty static going forward, and has a consistent naming scheme regarding underscores), I would propose instead collapsing the single- and multi-hue sequential gradients into a single group on the readme that could all be identified with the regex `^seq`.